packages/react-components/src/components/Picker/PickerList.tsx

```diff
@@ -0,0 +1,219 @@
+import * as React from 'react';
+import cx from 'clsx';
+import { Check } from '@livechat/design-system-icons/react/material';
+import { Icon, IconTypeName } from '../Icon';
+import styles from './PickerList.module.scss';
+import { KeyCodes } from '../../utils/keyCodes';
+import { TriggerSize } from './Trigger';
+
+const baseClass = 'picker-list';
+const itemClassName = `${baseClass}__item`;
+
+export interface IPickerListItem {
+  key: string;
+  name: string;
+  groupHeader?: boolean;
+  disabled?: boolean;
+}
+
+export interface IPickerListProps {
+  isOpen: boolean;
+  items: IPickerListItem[];
+  selectedItem: IPickerListItem | null;
+  size?: TriggerSize;
+  emptyStateText?: string;
+  onClose: () => void;
+  onSelect: (item: IPickerListItem) => void;
+}
+
+export const PickerList: React.FC<IPickerListProps> = ({
+  isOpen,
+  items,
+  selectedItem,
+  size = 'medium',
+  emptyStateText = 'No results found',
+  onClose,
+  onSelect,
+}) => {
+  const mergedClassNames = cx(
+    styles[baseClass],
+    styles[`${baseClass}--${size}`],
+    {
+      [styles[`${baseClass}__no-results`]]: items.length === 0,
+    }
+  );
+
+  const [currentItemKey, setCurrentItemKey] = React.useState<string | null>(
+    null
+  );
+  const indexRef = React.useRef(-1);
+  const lastIndexRef = React.useRef(0);
+  const listRef = React.useRef<HTMLUListElement>(null);
+
+  React.useEffect(() => {
+    indexRef.current = 0;
+
+    if (items.length > 0) {
+      setCurrentItemKey(items[indexRef.current].key);
+    }
+
+    if (isOpen) {
+      document.addEventListener('keydown', onKeyDown);
+    }
+
+    return () => document.removeEventListener('keydown', onKeyDown);
```

Comment:
If we register the event listener conditionally, then we should also return the cleanup function for its unregistration also conditionally 😉 

```suggestion
    if (isOpen) {
      document.addEventListener('keydown', onKeyDown);
      return () => document.removeEventListener('keydown', onKeyDown);
    }
```